crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -109,7 +135,15 @@ fn is_argument_only_used_in_recursion<'a>(
                     index == arg_index
                         && arg.span() == argument.span()
                         && if let Expression::Identifier(identifier) = &call_expr.callee {
-                            identifier.name == function_id.name
+                            identifier
+                                .reference_id()
+                                .and_then(|id| ctx.symbols().get_reference(id).symbol_id())
+                                .is_some_and(|v| {
+                                    v == function_id
+                                        .symbol_id
+                                        .get()
+                                        .expect("`symbol_id` should be set")
```

Comment:
Maybe use is_some_and alongside a debug_assert to avoid panics for release builds. I'm somewhat confident that this expectation will only panic when there's a bug in Semantic, but there's no reason for the linter to crash just because there's a bug.

Reply:
sounds good, fixed